radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokens/DelegatedTokenBalanceState.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the “Software”),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.radixdlt.client.application.translate.tokens;
+
+import com.google.common.collect.ImmutableMap;
+import com.radixdlt.client.application.translate.ApplicationState;
+import com.radixdlt.identifiers.RRI;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The tokens delegated to an address at a given point in time.
+ */
+public class DelegatedTokenBalanceState implements ApplicationState {
+	private final ImmutableMap<RRI, BigDecimal> balance;
+
+	public DelegatedTokenBalanceState() {
+		this.balance = ImmutableMap.of();
+	}
+
+	private DelegatedTokenBalanceState(Map<RRI, BigDecimal> balance) {
+		this.balance = ImmutableMap.copyOf(balance);
+	}
+
+	public Map<RRI, BigDecimal> getBalance() {
+		return balance;
+	}
+
+	public static DelegatedTokenBalanceState empty() {
+		return new DelegatedTokenBalanceState();
+	}
+
+	public static DelegatedTokenBalanceState combine(DelegatedTokenBalanceState state0, DelegatedTokenBalanceState state1) {
+		if (state0 == state1) {
```

Comment:
it seems a bit counterintuitive that this method won't merge an object with itself (by ref), while still merging two objects that are `equal`.

Reply:
I'm not sure why this is here, but I'm thinking it might be some kind of workaround for an issue in the upstream combiner.